exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/IterableEx.kt

```diff
@@ -26,6 +26,12 @@ interface SizedIterable<out T> : Iterable<T> {
     fun orderBy(vararg order: Pair<Expression<*>, SortOrder>): SizedIterable<T>
 }
 
+/** Represents the iterable elements of a database result, which are stored once loaded on first access. */
+interface LazySizedIterable<T> : SizedIterable<T> {
+    /** The lazily loaded database result. */
+    var loadedResult: List<T>?
+}
+
```

Comment:
An alternative to this would be to add a function to the base `SizedIterable` interface, something like
```kt
fun cacheLoadedResult(result: List<*>) {}
```
that could be overriden in the anonymous object below to set a private variable.

This issue with that would be that `cacheLoadedResult()` would be available to users for every SizedIterable (which means every single query, as `AbstractQuery` implements it):
```kt
MyTable.selectAll().cacheLoadedResult()
// this would do nothing and I'm not sure what the expected behavior/value should/could be
```